File: 1437_Check_If_All_1's_Are_at_Least_Length_K_Places_Away/solution.c
bool kLengthApart(int* nums, int numsSize, int k) {
    int* index = (int*)malloc(numsSize * sizeof(int));
    int indices = 0;

    for (int i = 0; i < numsSize; i++)
        if (nums[i] == 1) index[indices++] = i;
    for (int i = 0; i < indices - 1; i++)
        if (index[i + 1] - index[i] <= k) return false;
        
    return true;
}

File: 1437_Check_If_All_1's_Are_at_Least_Length_K_Places_Away/solution.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        index = []
        for i in range(len(nums)):
            if nums[i] == 1:
                index.append(i)

        for i in range(len(index) - 1):
            if index[i + 1] - index[i] <= k:
                return False

        return True